app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdScreen.kt

```diff
@@ -148,95 +136,41 @@ class ScanSimpleIdScreen : BaseScreen<
-
-    val analyzer = ImageAnalysis.Builder()
-        .setTargetAspectRatio(screenAspectRatio)
-        .setTargetRotation(rotation)
-        .build()
-
     val googlePlayServicesAvailability = googleApiAvailability.isGooglePlayServicesAvailable(requireContext())
     val isGooglePlayServicesAvailable = googlePlayServicesAvailability == ConnectionResult.SUCCESS
 
     val isMLKitQrCodeScannerEnabled = features.isEnabled(Feature.MLKitQrCodeScanner)
     val qrCodeAnalyzer = if (isMLKitQrCodeScannerEnabled && isGooglePlayServicesAvailable) {
       MLKitQrCodeAnalyzer(bitmapUtils, ::qrCodeScanned, mlKitUnavailable = {
-        setQrCodeAnalyzer(analyzer, ZxingQrCodeAnalyzer(::qrCodeScanned))
+        setQrCodeAnalyzer(ZxingQrCodeAnalyzer(::qrCodeScanned))
       })
     } else {
       ZxingQrCodeAnalyzer(::qrCodeScanned)
     }
 
-    setQrCodeAnalyzer(analyzer, qrCodeAnalyzer)
-
-    cameraProvider.unbindAll()
-
-    val camera = cameraProvider.bindToLifecycle(viewLifecycleOwner, cameraSelector, preview, analyzer)
-    preview.setSurfaceProvider(previewView.surfaceProvider)
-
-    previewView.setOnTouchListener { _, motionEvent ->
-      when (motionEvent.action) {
-        MotionEvent.ACTION_DOWN -> true
-        MotionEvent.ACTION_UP -> {
-          processTapToFocus(motionEvent, camera.cameraControl)
-          true
-        }
-        else -> false
-      }
-    }
+    setQrCodeAnalyzer(qrCodeAnalyzer)
   }
 
-  private fun setQrCodeAnalyzer(analyzer: ImageAnalysis, qrCodeAnalyzer: ImageAnalysis.Analyzer) {
-    analyzer.clearAnalyzer()
-    analyzer.setAnalyzer(cameraExecutor, qrCodeAnalyzer)
-  }
-
-  private fun processTapToFocus(motionEvent: MotionEvent, cameraControl: CameraControl) {
-    val meteringPointFactory = previewView.meteringPointFactory
-    val point = meteringPointFactory.createPoint(motionEvent.x, motionEvent.y)
-    val action = FocusMeteringAction.Builder(point).build()
-
-    cameraControl.startFocusAndMetering(action)
-  }
-
-  /**
-   *  [androidx.camera.core.ImageAnalysis.Builder.setTargetAspectRatio] requires enum value of
-   *  [androidx.camera.core.AspectRatio]. Currently it has values of 4:3 & 16:9.
-   *
-   *  Detecting the most suitable ratio for dimensions provided in @params by counting absolute
-   *  of preview ratio to one of the provided values.
-   *
-   *  @param width - preview width
-   *  @param height - preview height
-   *  @return suitable aspect ratio
-   *
-   *  source: https://github.com/android/camera-samples/blob/bfc46b68ef51f337104d3487c2394b204cf0d453/CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt#L350
-   */
-  private fun aspectRatio(width: Int, height: Int): Int {
-    val previewRatio = max(width, height).toDouble() / min(width, height)
-    if (abs(previewRatio - RATIO_4_3_VALUE) <= abs(previewRatio - RATIO_16_9_VALUE)) {
-      return AspectRatio.RATIO_4_3
-    }
-    return AspectRatio.RATIO_16_9
```

Comment:
Huh, have they incorporated this into the new library now?

Reply:
Not a new library, it's still in the same artifact. It's just they followed better SoC. One thing I am concerned about is there is no longer a way to set the aspect ratio. That's why I am on the fence about this implementation. 😅 

Requires thorough testing on all of our test devices.